source/slang/slang.cpp

```diff
@@ -3269,7 +3263,33 @@ void Module::updateDependencyBasedHash(
 
 void Module::updateContentsBasedHash(DigestBuilder& builder)
 {
-    builder.addToDigest(getContentsDigest());
+    auto filePathDependencies = getFilePathDependencies();
+
+    DigestBuilder lastModifiedBuilder;
+    for (auto file : filePathDependencies)
+    {
+        struct stat fileStatus;
+        stat(file.getBuffer(), &fileStatus);
+        lastModifiedBuilder.addToDigest(fileStatus.st_mtime);
```

Comment:
What if `stat` failed?

Reply:
hmm, safer to change this back to returning a `Result` then to help catch `stat` returning non-zero. I was thinking that the process of building up the list of file dependencies should catch any problems, but I suppose a file could be inaccessible even if it exists.

Reply:
I'm going to change the return type for `compute/updateDependencyBasedHash` to `Result` as well to maintain parity between the two functions (and in case any potential future additions to the hashing can fail).

Reply:
Changed to handle file dependencies which do not physically reside on disk.